pkg/crc/preflight/preflight_checks_tray_windows.go

```diff
@@ -4,95 +4,118 @@ import (
 	"github.com/code-ready/crc/pkg/os/windows/powershell"
-	"github.com/code-ready/crc/pkg/os/windows/secpol"
-	"github.com/code-ready/crc/pkg/os/windows/service"
 )
 
-var (
-	startUpFolder   = filepath.Join(constants.GetHomeDir(), "AppData", "Roaming", "Microsoft", "Windows", "Start Menu", "Programs", "Startup")
-	trayProcessName = constants.TrayBinaryName[:len(constants.TrayBinaryName)-4]
-)
+func checkIfTrayInstalled() error {
+	/* We want to force installation whenever setup is ran
+	 * as we want the service config to point to the binary
+	 * with which the setup command was issued.
+	 */
 
-func checkIfDaemonServiceInstalled() error {
-	// We want to force installation whenever setup is ran
-	// as we want the service config to point to the binary
-	// with which the setup command was issued
-	return fmt.Errorf("Ignoring check and forcing installation of daemon service")
+	return fmt.Errorf("Ignoring check and forcing installation of System Tray")
 }
 
-func fixDaemonServiceInstalled() error {
-	// only try to remove if a previous version exists
-	if service.IsInstalled(constants.DaemonServiceName) {
-		if service.IsRunning(constants.DaemonServiceName) {
-			_ = service.Stop(constants.DaemonServiceName)
-		}
-		_ = service.Delete(constants.DaemonServiceName)
+func fixTrayInstalled() error {
+	/* To avoid asking for elevated privilages again and again
+	 * this takes care of all the steps needed to have a running
+	 * tray by invoking a ps script which does the following:
+	 * a) Add logon as service permission for the user
+	 * b) Create the daemon service and start it
+	 * c) Add tray to startup folder and start it
+	 */
+
+	tempDir, err := ioutil.TempDir("", "crc")
+	if err != nil {
+		logging.Error("Failed creating temporay directory for tray installation")
+		return err
 	}
 
-	// get executables path
+	defer func() {
+		_ = goos.RemoveAll(tempDir)
+	}()
+	// prepare the ps script
 	binPath, err := goos.Executable()
 	if err != nil {
 		return fmt.Errorf("Unable to find the current executables location: %v", err)
 	}
 	binPathWithArgs := fmt.Sprintf("%s daemon", strings.TrimSpace(binPath))
 
-	// get the account name
-	u, err := user.Current()
-	if err != nil {
-		return fmt.Errorf("Failed to get username: %v", err)
-	}
-	logging.Debug("Got username: ", u.Username)
-	accountName := strings.TrimSpace(u.Username)
-
 	// get the password from user
 	password, err := input.PromptUserForSecret("Enter account login password for service installation", "This is the login password of your current account, needed to install the daemon service")
 	if err != nil {
 		return fmt.Errorf("Unable to get login password: %v", err)
 	}
-	err = service.Create(constants.DaemonServiceName, binPathWithArgs, accountName, password)
+
+	// escape '$' with '`' if present in password
+	// so that powershell doesn't try to exapnd it as variable
+	if strings.Contains(password, "$") {
+		password = strings.Replace(password, "$", "`$", -1)
```

Comment:
Is this the only character which needs escaping? What if the password contains a `? Or if the password is `"; somepowershellcommand`?

Reply:
We already include the string inside quotes, which escapes most of the things but not the `$`, but i need to test with the backquote and quotes..